include/pybind11/gil_safe_call_once.h

```diff
@@ -82,7 +85,11 @@ class gil_safe_call_once_and_store {
 private:
     alignas(T) char storage_[sizeof(T)] = {};
     std::once_flag once_flag_ = {};
+#ifdef Py_GIL_DISABLED
```

Comment:
Does it work on all platforms like this?

```diff
-    bool is_initialized_ = false;
+#ifdef Py_GIL_DISABLED
+    std::atomic_bool
+#else
+    bool
+#endif
+        is_initialized_{false};
```

Reply:
Yes, I think that should work.